### PR TITLE
Improve unit test coverage for azure publicipclient

### PR DESCRIPTION
**Pull request by v-xuxin**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by v-xuxin** [comment]
Signed

File: staging/src/k8s.io/legacy-cloud-providers/azure/clients/publicipclient/azure_publicipclient_test.go [hunk 1]
```diff
@@ -26,18 +26,100 @@ import (
...
+	defer ctrl.Finish()
+
+	pipGetErr := &retry.Error{
+		RawError:   fmt.Errorf("azure cloud provider throttled for operation %s with reason %q", "PublicIPGet", "client throttled"),
+		Retriable:  true,
+		RetryAfter: getFutureTime(),
+	}
+
+	armClient := mockarmclient.NewMockInterface(ctrl)
+
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/clients/publicipclient/azure_publicipclient_test.go:115]: nit: remove the empty line after armClient (same with other test cases)

File: staging/src/k8s.io/legacy-cloud-providers/azure/clients/publicipclient/azure_publicipclient_test.go [hunk 2]
```diff
@@ -80,6 +162,156 @@ func TestGetInternalError(t *testing.T) {
...
+	response := &http.Response{
+		StatusCode: http.StatusOK,
+		Body:       ioutil.NopCloser(bytes.NewReader([]byte("{}"))),
+	}
+	armClient := mockarmclient.NewMockInterface(ctrl)
+	armClient.EXPECT().GetResourceWithDecorators(gomock.Any(), resourceID, gomock.Any()).Return(response, nil).Times(1)
+	armClient.EXPECT().CloseResponse(gomock.Any(), gomock.Any()).Times(1)
+
+	pipClient := getTestPublicIPAddressClient(armClient)
+	expected := network.PublicIPAddress{Response: autorest.Response{Response: response}}
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/clients/publicipclient/azure_publicipclient_test.go:204]: could you add the actual IP address in expected struct?

**Comment by v-xuxin** [comment]
@feiskyer It seems that the job 'pull-kubernetes-e2e-kind' is not very stable, it always test failure.

**Pull request merged**